habushu-maven-plugin/src/main/java/org/technologybrewery/habushu/migration/CustomPoetrycoreVersionMigration.java

```diff
@@ -0,0 +1,162 @@
+package org.technologybrewery.habushu.migration;
+
+import com.electronwill.nightconfig.core.CommentedConfig;
+import com.electronwill.nightconfig.core.Config;
+import com.electronwill.nightconfig.core.file.FileConfig;
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.technologybrewery.baton.AbstractMigration;
+import org.technologybrewery.baton.BatonException;
+import org.technologybrewery.habushu.HabushuException;
+import org.technologybrewery.habushu.util.TomlReplacementTuple;
+import org.technologybrewery.habushu.util.TomlUtils;
+import org.technologybrewery.habushu.util.PoetryUtil;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+
+/**
+ * Automatically migrates poetry-core version (e.g., [poetry-core>=1.0.0]) in the
+ * [build-system] group to [poetry-core>=1.6.0] if the major version < 1 and minor version is < 6 as the
+ * Hashubu requires poetry-core version of 1.6.0 or higher. As noted in the
+ * project's README.md, this prevents these dependencies from causing issues when Poetry projects are exported in
+ * development releases.
+ */
+public class CustomPoetrycoreVersionMigration extends AbstractMigration {
+
+    public static final Logger logger = LoggerFactory.getLogger(CustomPoetrycoreVersionMigration.class);
+
+    protected Map<String, TomlReplacementTuple> replacements = new HashMap<>();
+
+    private boolean isMajorVersionUpdateRequired;
+    private boolean isMinorVersionUpdateRequired;
+
+    @Override
+    protected boolean shouldExecuteOnFile(File file) {
+        try (FileConfig tomlFileConfig = FileConfig.of(file)) {
+            tomlFileConfig.load();
+            Optional<Config> toolBuildSystem = tomlFileConfig.getOptional(TomlUtils.BUILD_SYSTEM);
+            if (toolBuildSystem.isPresent()) {
+                Config buildSystem = toolBuildSystem.get();
+                Map<String, Object> dependencyMap = buildSystem.valueMap();
+                for (Map.Entry<String, Object> dependency : dependencyMap.entrySet()) {
+                    isMajorVersionUpdateRequired = false;
+                    isMinorVersionUpdateRequired = false;
+                    // check if we need to upgrade the poetry-core version.
+                   if(IsPoetrycoreUpgradeRequired(dependency)) {
```

Comment:
A: Minor, the if statement is unnecessary.

Reply:
We are looping through the dependency map and iterate through all the elements to see if we any of it requires the upgrade. We need the if statement here incase any one of them requires upgrade.